recipe-client-addon/lib/EnvExpressions.jsm

```diff
@@ -5,13 +5,18 @@
 "use strict";
 
 const {utils: Cu} = Components;
+Cu.import("resource://gre/modules/Services.jsm");
 Cu.import("resource://gre/modules/XPCOMUtils.jsm");
 Cu.import("resource://gre/modules/TelemetryArchive.jsm");
 Cu.import("resource://gre/modules/Task.jsm");
 Cu.import("resource://shield-recipe-client/lib/Sampling.jsm");
 
+const {generateUUID} = Cc["@mozilla.org/uuid-generator;1"].getService(Ci.nsIUUIDGenerator);
+
 this.EXPORTED_SYMBOLS = ["EnvExpressions"];
 
+const prefs = Services.prefs.getBranch("extensions.shield-recipe-client.");
```

Comment:
I talked about it with Mythmon and we decided that the fail-fast behavior of `Services.prefs` is desirable in most places. We use Preferences.jsm in a few places where we _need_ defaults, like anything running before the spot where we set default pref values.